linux-tkg-config/prepare

```diff
@@ -662,457 +620,392 @@ CONFIG_DEBUG_INFO_BTF_MODULES=y\n
-    sed -i -e 's/CONFIG_GENERIC_CPU=y/# CONFIG_GENERIC_CPU is not set/' ./.config
-    echo "# CONFIG_GENERIC_CPU2 is not set" >> ./.config
-    echo "CONFIG_GENERIC_CPU3=y" >> ./.config
-    echo "# CONFIG_GENERIC_CPU4 is not set" >> ./.config
-  elif [ -n "$_processor_opt" ] && [ "$_processor_opt" = "genericv4" ]; then
-    sed -i -e 's/CONFIG_GENERIC_CPU=y/# CONFIG_GENERIC_CPU is not set/' ./.config
-    echo "# CONFIG_GENERIC_CPU2 is not set" >> ./.config
-    echo "# CONFIG_GENERIC_CPU3 is not set" >> ./.config
-    echo "CONFIG_GENERIC_CPU4=y" >> ./.config
-  else
-    echo "# CONFIG_GENERIC_CPU2 is not set" >> ./.config
-    echo "# CONFIG_GENERIC_CPU3 is not set" >> ./.config
-    echo "# CONFIG_GENERIC_CPU4 is not set" >> ./.config
-  fi
-
-  _cpu_marchs=("k8" "k8sse3" "k10" "barcelona" "bobcat" "jaguar" "bulldozer" "piledriver")
+  _cpu_marchs=("native_amd" "native_intel" "generic_cpu" "generic_cpu2" "generic_cpu3" "generic_cpu4")
+  _cpu_marchs+=("k8" "k8sse3" "k10" "barcelona" "bobcat" "jaguar" "bulldozer" "piledriver")
   _cpu_marchs+=("steamroller" "excavator" "zen" "zen2" "zen3" "mpsc" "atom" "core2" "nehalem" "westmere")
   _cpu_marchs+=("bonnell" "silvermont" "sandybridge" "ivybridge" "haswell" "broadwell" "skylake")
   _cpu_marchs+=("skylakex" "cannonlake" "icelake" "goldmont" "goldmontplus" "cascadelake")
   _cpu_marchs+=("cooperlake" "tigerlake" "sapphirerapids" "rocketlake" "alderlake")
-  _cpu_marchs+=("native" "native_amd" "native_intel")
-
-  if [ ! -z "$_processor_opt" ]; then # Don't populate when the config variable is empty
-    for _march in "${_cpu_marchs[@]}"
-    do
-      _march_upper=`echo ${_march} | tr '[:lower:]' '[:upper:]'`
-      if [ "$_processor_opt" = "$_march" ]; then
-        echo "CONFIG_M${_march_upper}=y/" >> ./.config
+
+  typeset -A _generic_march_map
+  _generic_march_map=(
+    ["generic"]="generic_cpu"
+    ["generic_v2"]="generic_cpu2"
+    ["generic_v3"]="generic_cpu3"
+    ["generic_v4"]="generic_cpu4"
+  )
+  if [ -n "$_processor_opt" ]; then
+    # Replace customization.cfg convention with .config convention for generic
+    if [[ "${!_generic_march_map[@]}" =~ "$_processor_opt" ]]; then
+      _processor_opt="${_generic_march_map[$_processor_opt]}"
+    fi
+    if ! [[ "${_cpu_marchs[@]}" =~ "$_processor_opt" ]]; then
+      warning "the setup _processor_opt=\"${_processor_opt}\" not recognized. Prompting..."
+      _processor_opt=""
+    fi
+  fi
+
+  if [ -z "$_processor_opt" ]; then
+    msg2 "Please select the desired CPU micro-architecture"
+    _default_index="2"
+    _prompt_from_array "${_cpu_marchs[@]}"
+    _processor_opt="${_selected_value}"
+  fi
+
+  for _march in "${_cpu_marchs[@]}"
+  do
+    if [ "$_processor_opt" = "$_march" ]; then
+      if [[ "$_march" == "generic"* ]]; then
+        _enable "${_march}"
       else
-        echo "# CONFIG_M${_march_upper} is not set" >> ./.config
+        _enable "M${_march}"
       fi
-    done
-  fi
+    else
```

Comment:
You'll also need to handle the generic marchs in the disable side, else we're getting prompted even though we have selected the march we wanted already:
```
  for _march in "${_cpu_marchs[@]}"
  do
    if [ "$_processor_opt" = "$_march" ]; then
      if [[ "$_march" = "generic"* ]]; then
        _enable "${_march}"
      else
        _enable "M${_march}"
      fi
    else
      if [[ "$_march" = "generic"* ]]; then
        _disable "${_march}"
      else
        _disable "M${_march}"
      fi
    fi
  done
```
(taking the _undefine -> _disable change into account)

Reply:
Yes I forgot about it x) Thanks!